docs/ilab-model-engine.md

```diff
@@ -0,0 +1,201 @@
+
+An alternative flow to `ilab` -> `child-command` -> `grandchild-command` is:
+
+`ilab` -> `verb` -> `noun`. The positive to this approach is commands would sound better: `ilab generate data`. However there are a few negatives. While this is easier to say, it makes less sense from an organizational standpoint:
+
+```console
+ilab
+|
+|_______download__|
+|_______convert___|-----model
+|_______train_____|
+```
+
+This structure works when commands are ALL linked to the same `verb`. What happens when a group like `download` gets a command that the others don't? Then it looks more like this:
+
+```console
+ilab
+|
+|_______download
+|       |____model
+|       |____taxonomy
+|_______convert
+|       |_____model
+|_______train
+|       |_____model
+```
+
+This is opposed to the structure in this EP, which might have a duplicate here and there for something like `download`, but in this structure duplicate commands are the norm and will result in a larger and clunkier codebase that is confusing to read.
+
+This first structure looks nice, and `ilab download model` sounds nice. However in terms of implementation, this makes little sense. in [click](https://click.palletsprojects.com/en/8.1.x/), the cli library we use, `ilab` is a "group" that commands are currently "grouped under". In this Model, `ilab` would be a group, `download`, `convert` etc would be a sub-group, and `model` would be a command under ALL of these groups. This would require an implementation of the `model` function in different packages all linking to a different group parent. From the perspective of user contributions and general code readability doesn't make much sense. Part of this design is to encourage user contributions by making the structure of the codebase logical. While these commands make the commands sound better, they make them harder to group, and understand.
+
+### General workflow as compared to alternatives
+
+currently one has to:
+
+1. `ilab init`
+2. `ilab download`
+3. `ilab serve`
+4. `ilab generate`
+5. `ilab train`
+6. `ilab convert`
+7. `ilab serve --model XXX`
+8. `ilab chat`
+
+how to use `ilab` with this new structure:
+
+1. `ilab config init`
+2. `ilab model download`
+3. `ilab data generate`
+4. `ilab model train --convert`
+5. `ilab model serve -i` (or followed by `ilab model chat`)
+
+So, not only we do remove one of the steps (with -i) for most users, but it is actually clear what is happening. For example one may ask: What does `ilab init` do? While `ilab config init` clearly is initializing the config of the cli. A more clear example even is: `ilab generate`. Are we generating a model? A config? No we are generating *data*: `ilab data generate`.
+
+## Changes to Existing flow
+
+The current `ilab` commands will still work. Users will be able to type commands like `ilab model train` or `ilab train` for the foreseeable future to ensure feature parity. Eventually, this alias should be removed, and only the sub-commands should probably exist.
```

Comment:
We should target an explicit release to do so (outside of this general proposal, in our usual planning meetings/channels)

Reply:
we'll have to revisit having 2 ways of doing everything for sure.  I guess it is comforting to suggest we a dual path for transition.